Set SupportedVersion on GatewayClass (#1301)
nginxinc/nginx-gateway-fabric: docs/developer/release-process.md

3. Test the main branch for release-readiness. For that, use the `edge` containers, which are built from the main
   branch, and the [example applications](/examples).
4. If a problem is found, prepare a fix PR, merge it into the main branch and return to the previous step.

> 5. If the supported Gateway API minor version has changed since the last release, test NGF with the previous version of the Gateway API CRDs.

Reviewer: ```suggestion
5. If the supported Gateway API minor version has changed since the last release, test NGINX Gateway Fabric with the previous version of the Gateway API CRDs.
```

6. If a problem is found, add a note to the release notes explaining that the previous version is not supported.

> 7. Create a release branch with a name that follows the `release-X.Y` format.

Reviewer: ```suggestion
7. Create a release branch following the  `release-X.Y` naming convention.
```
